bluesky/plans.py

```diff
@@ -148,14 +198,147 @@ def rel_list_scan(detectors, motor, steps, *, per_step=None, md=None):
     _md = {'plan_name': 'rel_list_scan'}
     _md.update(md or {})
 
-    @bpp.reset_positions_decorator([motor])
-    @bpp.relative_set_decorator([motor])
+    motors= [motor for motor, pos_list in partition(2, args)]
+
+    @bpp.reset_positions_decorator(motors)
+    @bpp.relative_set_decorator(motors)
     def inner_relative_list_scan():
-        return (yield from list_scan(detectors, motor, steps,
-                                     per_step=per_step, md=_md))
+        return (yield from list_scan(detectors, *args, per_step=per_step,
+                                     md=_md))
     return (yield from inner_relative_list_scan())
 
 
+def list_grid_scan(detectors, *args, per_step=None, md=None):
+    """
+    Scan over a mesh; each motor is on an independent trajectory.
+
+    Parameters
+    ----------
+    detectors : list
+        list of 'readable' objects
+
+    args
+        patterned like (``motor1, position_list1,``
+                        ``motor2, position_list2,``
+                        ``motor3, position_list3,``
+                        ``...,``
+                        ``motorN, position_listN``)
+
+        The first motor is the "slowest", the outer loop. ``position_list``'s
+        are a list of positions, all lists must have the same length. Motors
+        can be any 'settable' object (motor, temp controller, etc.).
+
+    snake_axes
+        which axes should be snaked, either None (do not snake any axes),
```

Comment:
We settled on `False` for this, though if the user passes in `None` it will have the same result as `False`. See [comment](https://github.com/bluesky/bluesky/issues/1189#issuecomment-492872715) and discussion below. (The default of `False`  well suggests to the rushed reader that `True` might be a valid option.)

Reply:
fixed